pytorch_lightning/trainer/trainer.py

```diff
@@ -913,12 +913,13 @@ def tune(
 
     def _restore_modules_and_callbacks(self) -> None:
         # restore modules after setup
+        self.checkpoint_connector.resume_start()
+
         if self.state.fn == TrainerFn.FITTING:
-            self.checkpoint_connector.resume_start()
-        self.checkpoint_connector.restore_datamodule()
-        self.checkpoint_connector.restore_model()
-        # restore callback states
-        self.checkpoint_connector.restore_callbacks()
+            self.checkpoint_connector.restore_datamodule()
+            self.checkpoint_connector.restore_model()
+            # restore callback states
+            self.checkpoint_connector.restore_callbacks()
```

Comment:
FYI: this is required to at least load the resume_from_checkpoint and assign global_step and epoch during val/test as suggested by Thomas above. `reload_datamodule` and `restore_callbacks` are now reloaded during `.fit`. Please let me know whether it's the intended behavior or not.